build/testing/integration.go

```diff
@@ -72,14 +92,26 @@ type authConfig int
 
 const (
 	noAuth authConfig = iota
-	authNoNamespace
-	authNamespaced
+	staticAuth
+	staticAuthNamespaced
+	jwtAuth
```

Comment:
When you said yesterday that the ITs were driving you up the wall I now know it was this bit 😂

Reply:
One thing, just to say outloud, that we don't have to make it a part of this PR (to avoid scope creep):

Ultimately, I think we want to circle back to these ITs being as they currently are (authNoNamespace and authNamespaced, not shenagans with JWT vs token).
Instead of JWT not supporting namespace, we just ultimately have a namespace claim in the JWT somewhere (I think we should have top-level key (e.g. `flipt`) and then put the rest of our policy beneath that).

In that world, instead of this causing more complex branching and conditions, all the current conditions hold true still, we just setup a test case for JWT with and without a namespace scope.

That said, supporting this will have some refactoring implications. Particularly around the fact that the current namespace middleware is static token hard-coded:
https://github.com/flipt-io/flipt/blob/5aa70ec41aa67e0e5e2605ba99ec18283a1c9b92/internal/server/auth/middleware/grpc/middleware.go#L211C22-L225

If we did this, I would suggest we enapsulate extracting the namespaces behind a method or put the authenticaiton behind an interface.

Again, PR size tradeoff. We could either add support for namespaced JWT in order to reduce the branching complexity in the ITs today. Or we can skip it and have all these conditions for now. Then add it later and put it back.